api/v1alpha1/watermarkpodautoscaler_types.go

```diff
@@ -204,3 +205,30 @@ type WatermarkPodAutoscalerList struct {
 func init() {
 	SchemeBuilder.Register(&WatermarkPodAutoscaler{}, &WatermarkPodAutoscalerList{})
 }
+
+// GetLogAttrs takes a list of log attrs that would be sent to logger.info, and adds user specified key value attribute
+// pairs from the ad.datadoghq.com/attributes annotation of the WPA spec.
+// logAttrs should be pairs of key/values, with each key being a string.
+// This func returns a slice! To pass this as variadic parameters to logger.info, you need the ellipses, so use wpa.GetLogAttrs()...
+func (wpa *WatermarkPodAutoscaler) GetLogAttrs(logAttrs ...interface{}) []interface{} {
```

Comment:
I would prefer to have a function in the `controllers` package like:

```golang
func GetLogAttrsFromWPA(wpa *apiv1alpha1.WatermarkPodAutoscaler) []interface{} {
  // ...
}
```

because the `v1alpha1` package is used to generate CRD definition.
So I would prefer that we keep it only with resource definition